src/main/java/com/google/devtools/build/lib/remote/ByteStreamUploader.java

```diff
@@ -83,7 +89,7 @@
   private final Set<HashCode> uploadedBlobs = new HashSet<>();
 
   @GuardedBy("lock")
-  private final Map<Digest, ListenableFuture<Void>> uploadsInProgress = new HashMap<>();
+  private final Map<HashCode, ListenableFuture<Void>> uploadsInProgress = new HashMap<>();
```

Comment:
Why do you need to change the types here?

Reply:
Once I'd implemented the progressive upload, I found I had no need for the digest size, and factored it out. It also makes ByteStreamUploader independent of REAPI, similar to how progressive reads made AbstractRemoteActionCache independent of gRPC.